packages/algjulia-service/src/AlgebraicJuliaService.jl

```diff
@@ -7,27 +7,72 @@ using Reexport
 # the intent is that this is an interface for AlgebraicJulia code to interoperate with CatColab 
 @data AlgebraicJuliaIntegration begin
     ThDecapode()
-end
-export ThDecapode
+end; export ThDecapode
+
+# cells in the JSON are tagged. these are just objects for dispatching `to_model`
+@data ElementTag begin
+    ObTag()
+    HomTag()
+end; export ObTag, HomTag
+
+#=
+@active patterns are MLStyle-implementations of F# active patterns that forces us to work in the Maybe/Option pattern. 
+Practically, yet while a matter of opinion, they make @match statements cleaner; a statement amounts to a helpful pattern
+name and the variables we intend to capture.
+=# 
+@active IsObject(x) begin; x[:tag] == "object" ? Some(x) : nothing end
+@active IsMorphism(x) begin; x[:tag] == "morphism" ? Some(x) : nothing end
+export IsObject, IsMorphism
+
+""" Obs, Homs """
+abstract type ModelElementValue end
```

Comment:
Why use an abstract type here instead of `@data` as used for `ElementTag` above? Also, I'd suggest a more uniform naming schema, such as:

- `ElementTag` -> `ModelElementTag`
- `ModelElementValue` -> `ModelElement`

Reply:
I've changed them to use the `@data` macro, and renamed `ElementTag` -> `ModelElementTag`, but I've retained the naming `ModelElement` so its `val` parameter is still parameterized over something named `ModelElementValue`